packages/internal/src/gql.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  visit,
+  DocumentNode,
+  OperationTypeNode,
+  OperationDefinitionNode,
+  FieldNode,
+} from 'graphql'
+
+interface Operation {
+  operation: OperationTypeNode
+  name: string | undefined
+  fields: Array<string | Field>
+}
+
+interface Field {
+  string: Array<string | Field>
+}
+
+export const parseDocumentAST = (document: DocumentNode) => {
+  const operations: Array<Operation> = []
+
+  visit(document, {
+    OperationDefinition(node: OperationDefinitionNode) {
+      const fields: any[] = []
+
+      node.selectionSet.selections.forEach((field) => {
+        fields.push(getFields(field as FieldNode))
+      })
+
+      operations.push({
+        operation: node.operation,
+        name: node.name?.value,
+        fields,
+      })
+    },
+  })
+
+  return operations
+}
+
+const getFields = (field: FieldNode): any => {
+  // base
+  if (!field.selectionSet) {
+    return field.name.value
+  } else {
+    const obj = {
```

Comment:
```suggestion
    const obj: Record<string, FieldNode[]>
```
This fixes the type error :)